core/state_processor.go

```diff
@@ -337,13 +323,21 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc *BlockCh
 // and uses the input parameters for its environment. It returns the receipt
 // for the transaction, gas used and an error if the transaction failed,
 // indicating the block was invalid.
-func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb, privateStateDB *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, error) {
+func ApplyTransaction(config *params.ChainConfig, bc *BlockChain, author *common.Address, gp *GasPool, statedb, privateStateDB *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config, forceNonParty bool) (*types.Receipt, *types.Receipt, error) {
```

Comment:
TODO: we can continue to use ChainContext for bc, but we need to add a function on that interface.

Reply:
what is geth using?

Reply:
ChainContext, so I try to use that one then

Reply:
With ChainContext, it looks like this: https://github.com/ConsenSys/quorum/pull/1223/commits/ae50ace84f48b1a587abac5b4bb4ff7dfdef24e4